app/src/main/java/com/example/grocerieswizard/ui/home/HomePresenter.java

```diff
@@ -0,0 +1,76 @@
+package com.example.grocerieswizard.ui.home;
+
+import com.example.grocerieswizard.data.local.model.CartItem;
+import com.example.grocerieswizard.data.repo.RecipeRepository;
+import com.example.grocerieswizard.ui.UiMapper;
+import com.example.grocerieswizard.ui.model.RecipeUi;
+
+import java.util.stream.Collectors;
+
+public class HomePresenter implements HomeContract.Presenter {
+    private HomeContract.View view;
+    private final RecipeRepository recipeRepository;
+    private final UiMapper uiMapper;
+
+    public HomePresenter(RecipeRepository recipeRepository, UiMapper uiMapper) {
+        this.recipeRepository = recipeRepository;
+        this.uiMapper = uiMapper;
+    }
+
+    public void bindView(HomeContract.View view){
+        if(view!=null){
+         this.view = view;
+        }
+    }
+    public void unbindView(HomeContract.View view){
```

Comment:
This should be in the HomeContract.Presenter as well. And it doesn't need argument.